### PR TITLE
Initial version of homing missile + dart modifications

### PR DESCRIPTION
**Pull request by martkartasev**
Things to address 

- [x] Naming (Homing  vs Targeted)
- [x] Versions for AoE spells ?
- [ ] Restoring Original dart ?
- [ ] Adding Crosshair support for player.
- [x] Better way of passing target to script?
- [x] Min / Max / Disable distances.

### CODE REVIEW

File: CSharpSourceCode/Abilities/SeekerController.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                var error = _target.Position + new Vec3(0, 0, 2) - particleDirection;
+                if (error.Length < _parameters.DisableDistance)
+                {
+                    enabled = false;
+                    return globalFrame;
+                }
+
+                if (error.Length < _parameters.MaxDistance && error.Length > _parameters.MinDistance)
+                {
+                    var correction = error * _parameters.Proportional + (error - _prevError) * _parameters.Derivative;
```
Comment (Z3rca) [inline CSharpSourceCode/Abilities/SeekerController.cs:35]: Potentially the line I need a little more explaination... just from the variable names or the calculation I can't really understand the correction.

**Review by Z3rca — APPROVED** [review]
no real objections with this

**Review by hunharibo — APPROVED** [review]
No real comments from me. Awesome job.

**Comment by hunharibo** [comment]
I am not sure that we need the original version of Dart, i like the homing version better.

**Pull request merged**